Api/Controllers/PostController.cs

```diff
@@ -14,30 +16,54 @@ namespace Api.Controllers;
 public class PostController : ControllerBase
 {
     private IPostService postService;
+    private IUserService userService;
 
-    public PostController(IPostService postService)
+    public PostController(IPostService postService, IUserService userService)
     {
         this.postService = postService;
+        this.userService = userService;
+    }
+
+    [HttpGet]
+    [Authorize]
+    [Route("{id}")]
+    [SwaggerOperation("Get post details")]
+    public ActionResult<PostDto> GetPostDetails([FromRoute] Guid id)
+    {
+        var post = postService.AsUser(User.GetGuid()).GetPost(id);
+        return Ok(post.ToDto());
     }
 
     [HttpGet]
     [Authorize]
     [Route("user")]
     [SwaggerOperation("List user's subscribed events' posts")]
-    public ActionResult<Paged<PostDto>> GetUserEventsPosts([FromQuery] Paging paging, [FromQuery] bool showAlsoInterested)
+    public ActionResult<Paged<ListPostDto>> GetUserEventsPosts([FromQuery] Paging paging, [FromQuery] bool showFrindsPosts = false)
     {
-        var posts = postService.AsUser(User.GetGuid()).GetUsersPosts();
-        if (!showAlsoInterested)
-            posts = posts.FindAll(p => p.Event.Participants.FirstOrDefault(part => part.Guid == User.GetGuid()) is not null);
-        return Paged<PostDto>.PageFrom(posts.Select(p => p.ToDto()), DTO.Posts.CreationDateComparer.Instance, paging);
+        var user = userService.AsUser(User.GetGuid()).GetUser();
```

Comment:
It is ok, but just in my opinion it would be better to call postService.AsUser().ActingUser so that we dont inject UserService there.